route/src/route.cpp

```diff
@@ -63,8 +63,12 @@ namespace route {
     void Route::run()
     {
         initialize();
+
         // spin with spin_callback function from RouteGeneratorWorker
-        ros::CARMANodeHandle::setSpinCallback(std::bind(&RouteGeneratorWorker::spin_callback, &rg_worker_));
+        ros::Timer discovery_pub_timer_ = pnh_->createTimer(
```

Comment:
`discovery_pub_timer` is probably not the best name for this, since there's no plugin discovery going on here. `spin_timer`?

Reply:
Done